src/tests/CMakeLists.txt

```diff
@@ -16,5 +16,96 @@ endif()
+                        RESULT_VARIABLE GIT_SUBMOD_RESULT)
+        if(NOT GIT_SUBMOD_RESULT EQUAL "0")
+            message(FATAL_ERROR "git submodule update --init failed with ${GIT_SUBMOD_RESULT}, please checkout submodules")
+        endif()
+
+        # Define submodule tag
+        execute_process(COMMAND ${GIT_EXECUTABLE} checkout ${ANTARES_TEST_TAG}
+                        WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}/resources/Antares_Simulator_Tests
+                        RESULT_VARIABLE GIT_SUBMOD_RESULT)
+        if(NOT GIT_SUBMOD_RESULT EQUAL "0")
+            message(FATAL_ERROR "git checkout ${ANTARES_TEST_TAG} failed with ${GIT_SUBMOD_RESULT}, please checkout submodules")
+        endif()
+        
+        macro(examples_tests use_ortools ortools_solver)
+        
+            if(${use_ortools})
+                SET(SHORT_EXAMPLES_NAME short-examples-ortools-${ortools_solver})
+                SET(MEDIUM_EXAMPLES_NAME medium-examples-ortools-${ortools_solver})
+                SET(LONG_EXAMPLES_NAME long-examples-ortools-${ortools_solver})
+                
+                set(ADDITIONNAL_LABELS ortools ${ortools_solver})
+                
+                set(${use_ortools_option} "--use-ortools")
+            else()
+                SET(SHORT_EXAMPLES_NAME short-examples-sirius)
+                SET(MEDIUM_EXAMPLES_NAME medium-examples-sirius)
+                SET(LONG_EXAMPLES_NAME long-examples-sirius)
+                                
+                set(ADDITIONNAL_LABELS sirius)
+                
+                set(${use_ortools_option} "")
+                
+            endif()
+        
+            # Short tests
+            add_test(
+                NAME ${SHORT_EXAMPLES_NAME}
+                COMMAND Python3::Interpreter -m pytest -m short --solver-path=$<TARGET_FILE:antares-${ANTARES_PRG_VERSION}-solver> ${use_ortools_option} --ortools-solver=${ortools_solver} example_test.py
+                WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/examples"
+            )
+            set_property(TEST ${SHORT_EXAMPLES_NAME} PROPERTY LABELS short-examples ${ADDITIONNAL_LABELS})
+
+            # Medium tests
+            add_test(
+                NAME ${MEDIUM_EXAMPLES_NAME}
+                COMMAND Python3::Interpreter -m pytest -m medium --solver-path=$<TARGET_FILE:antares-${ANTARES_PRG_VERSION}-solver> ${use_ortools_option} --ortools-solver=${ortools_solver} example_test.py
+                WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}/examples"
+            )
+            set_property(TEST ${MEDIUM_EXAMPLES_NAME} PROPERTY LABELS medium-examples ${ADDITIONNAL_LABELS})
+
+            # Long tests
+            add_test(
+                NAME ${LONG_EXAMPLES_NAME}
+                COMMAND Python3::Interpreter -m pytest -m long --solver-path=$<TARGET_FILE:antares-${ANTARES_PRG_VERSION}-solver> ${use_ortools_option} --ortools-solver=${ortools_solver} example_test.py
```

Comment:
I guess **solver-path** option value is passed into **example_test.py** as **solver_path** via **conftest.py**, isn't it ?
Otherwise, how does **example_test.py** know about use_**ortools**, **ortools_solver**, **solver_path** ?

Reply:
Exactly, AFAIK there is no easy way to determine these, especially when multiple build directories co-exist